src/indexers/governance-proposals-indexer.ts

```diff
@@ -0,0 +1,329 @@
+import { GovernanceProposalsStatuses } from '../aws/models/governance-proposals-statuses.interface';
+import { getSupportedChains } from '../chains/chains.utils';
+import { Chain } from '../chains/config/chain.config';
+import { getLastProposalUpdateBlock } from '../governance/governance.utils';
+
+export async function updateGovernanceProposals() {
+  console.info('Updating governance proposals started');
+
+  const mapper = getDataMapper();
+
+  for (const chain of getSupportedChains()) {
+    const timelockAddress = chain.sdk.governance.timelockAddress;
+
+    if (!timelockAddress) {
+      console.info(`No timelock address for ${chain.network}`);
+      continue;
+    }
+
+    const lastScannedBlock = await getLastProposalUpdateBlock(chain.network);
+
+    const governanceProxy = new GovernanceProxyMock(chain.sdk);
+
+    const scanRangeOpts = {
+      startBlock: 0,
+      endBlock: 0,
+    };
+
+    // in sake of MVP, arbitrum gens blocks too fast, so we need to limit the range
+    // otherwise this schedulle won't complete in hours for the 1st run
+    if (!lastScannedBlock) {
+      await governanceProxy.processEventsScanRangePr(timelockAddress, scanRangeOpts);
+      if (scanRangeOpts.endBlock - scanRangeOpts.startBlock > 2_000_000) {
+        scanRangeOpts.startBlock = scanRangeOpts.endBlock - 2_000_000 + 1;
+      }
+    } else {
+      scanRangeOpts.startBlock = lastScannedBlock + 1;
+    }
+
+    const proposalsCreated = await chain.sdk.governance.loadScheduledProposals(scanRangeOpts);
+    const proposalsStatusesChanged = await chain.sdk.governance.loadProposalsStatusChange(scanRangeOpts);
+    const proposalsDisputed = await chain.sdk.governance.loadProposalsDisputes(scanRangeOpts);
+
+    if (proposalsCreated.length === 0 && proposalsStatusesChanged.length === 0 && proposalsDisputed.length === 0) {
+      console.info(`No new proposals for ${chain.network}`);
+      continue;
+    }
+
+    const updatedProposalsStatuses: (CallExecutedEvent | CancelledEvent)[] = [];
+    const updatedProposalsDisputes: (CallDisputedEvent | CallDisputedResolvedEvent)[] = [];
+
+    // save new created proposals
+    const updatedProposals: GovernanceProposals[] = [];
+
+    for (const proposal of proposalsCreated) {
+      const isBatchChild = proposal.args.index.toNumber() > 0;
+
+      if (!isBatchChild) {
+        const { proposalModel, statuses, disputes } = await saveRelativeProposal(
+          chain,
+          timelockAddress,
+          proposal,
+          proposalsStatusesChanged,
+          proposalsDisputed,
+        );
+        updatedProposals.push(proposalModel);
+        updatedProposalsStatuses.push(...statuses);
+        updatedProposalsDisputes.push(...disputes);
+        continue;
+      }
+
+      await saveChildProposal(chain, timelockAddress, proposal, updatedProposals, mapper);
+    }
+
+    try {
+      updatedProposals.length > 0 && (await mapper.put(updatedProposals));
```

Comment:
should this be batch put?

Reply:
i wonder this API all the time, ye you are right, thanks